src/include/random.h

```diff
@@ -141,6 +153,11 @@ int generate_random_number(MutexT& m, EngineT& e)
 
 } // namespace detail
 
+bool rng_is_seeded() 
+{
+  return detail::rng_is_seeded_state();
+}
```

Comment:
> checking a flag for each number you want, every time

note that, at least for the thread-local interfaces, this check doesn't need to be atomic